src/core/src/engine/mod.rs

```diff
@@ -61,10 +103,20 @@ impl Engine for Poet2Engine {
     ) -> Result<(), Error> {
         info!("Started PoET 2 Engine...");
 
-        let validator_id = Vec::from(startup_state.local_peer_info.peer_id);
+        // Register if it's not genesis node
+        if self.config.is_genesis() == false {
+            submit_batchlist_to_rest_api(
+                self.config.get_rest_api().as_str(),
+                BATCHES_REST_API,
+                self.batch_list.clone(),
+            );
+        }
+
+        let validator_id = self.validator_id.clone();
+
         let chain_head = startup_state.chain_head;
 
-        let mut service = Poet2Service::new(service);
+        let mut service = Poet2Service::new(service, self.enclave.clone());
```

Comment:
Ideally, either engine or service should maintain enclave. Need refactoring from service to handle this better.